src/test/java/com/juniorstart/juniorstart/controller/UserControllerTest.java

```diff
@@ -1,39 +1,187 @@
 package com.juniorstart.juniorstart.controller;
 
+import com.google.gson.Gson;
+import com.juniorstart.juniorstart.exception.ResourceNotFoundException;
+import com.juniorstart.juniorstart.model.audit.UserStatus;
+import com.juniorstart.juniorstart.payload.ApiResponse;
 import com.juniorstart.juniorstart.payload.ChangeMailRequest;
+import com.juniorstart.juniorstart.payload.ChangePasswordRequest;
+import com.juniorstart.juniorstart.payload.ChangeStatusRequest;
 import com.juniorstart.juniorstart.service.UserService;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.*;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.InjectMocks;
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
 import org.springframework.test.web.servlet.MockMvc;
-import org.springframework.test.web.servlet.MockMvcBuilder;
+import org.springframework.test.web.servlet.MvcResult;
 import org.springframework.test.web.servlet.setup.MockMvcBuilders;
 
-import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.core.Is.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/** Test for UserController class.
+ * @author Noboseki
+ * @version 1.0
+ * @since 1.0
+ */
 @ExtendWith(MockitoExtension.class)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
 class UserControllerTest {
     @Mock
     UserService userService;
-
-    @Mock
-    ChangeMailRequest changeMailRequest;
-
     @InjectMocks
     UserController userController;
 
+    Gson gson = new Gson();
     MockMvc mockMvc;
 
+    ResponseEntity<ApiResponse> response = ResponseEntity.ok((new ApiResponse(true,"Test Ok")));
+    ChangePasswordRequest changePasswordRequest = new ChangePasswordRequest("Password 2",UUID.randomUUID(), "Password");
+    ChangeMailRequest changeMailRequest = new ChangeMailRequest("test@tes.com", UUID.randomUUID(), "Password");
```

Comment:
Inicjalizacja zmiennych powinna być w odzielnej metodzie np w setUp

Reply:
A po co.  Jak nazwa wskazuje before Each mówi że przed każdym testem inicjalizowana jest zmienna. Ja uważam że to nie jest potrzebne żeby te zmienna inicjalizować za każdym razem przed użyciem, jeżeli nie jest potrzebna.

Reply:
Spoko